src/app.ts

```diff
@@ -165,20 +175,64 @@ export default (
       method: 'updateSecret',
     });
 
+    // I think for now we can use the taskName Prefix and edit it when we
+    // are adding additional information to the check
+    const checkResponse = await context.octokit.checks.create({
+      owner: context.payload.organization.login,
+      repo: '.github',
+      name: 'peribolos-run',
+      head_sha: context.payload.after,
+      status: 'queued',
+    });
+
     // Trigger taskrun to apply config changes to org
-    wrapOperationWithMetrics(
-      useApi(APIS.CustomObjectsApi).createNamespacedCustomObject(
-        'tekton.dev',
-        'v1beta1',
-        getNamespace(),
-        'taskruns',
-        generateTaskPayload('peribolos-run', context)
-      ),
+    createTaskRun('peribolos-run', context, [
       {
-        install: context.payload.installation.id,
-        method: 'schedulePushTask',
-      }
-    );
+        name: 'CHECK_RUN_ID',
+        value: checkResponse.data.id.toString(),
+      },
+    ]);
+  });
+
+  app.on('check_run.rerequested', async (context: any) => {
+    // In the future a check is needed if this is the peribolos-run check
+    const checkCommit = context.payload.check_run.head_sha;
+    const comRepo = await context.octokit.repos.get({
+      owner: context.payload.organization.login,
+      repo: '.github',
+    });
+    const defaultBranch = await context.octokit.repos.getBranch({
+      owner: context.payload.organization.login,
+      repo: '.github',
+      branch: comRepo.data.default_branch,
+    });
+    const headCommit = defaultBranch.data.commit.sha;
+
+    if (checkCommit === headCommit) {
```

Comment:
A tiny nit to improve the code style:

In general, when you're branching `if else` you want to avoid long spaghetti code and deep and long nesting so you don't loose context when reading the code. Therefore you should strive as short `if else` bodies as possible. For example here it's much easier to follow if you flip the if statement - in the `if` body you can complete the resolution fast and `return` and then continue without being nested in `else`... If we need to extend this particular code in future, it's more straightforward and easier to find the right spot where the new code should go.

```ts
if (checkCommit !== headCommit) {
  # set check to skipped
  return # exit the handler, because we don't want any further processing
}
# no else needed because we return above, hence no nesting
...
# set check to queued
# schedule taskrun
```

Reply:
Thanks for the detailed explanation, I will definitely use this more in the future!